src/main/java/org/wildfly/naming/client/util/NamingUtils.java

```diff
@@ -57,7 +57,7 @@ public static CompositeName toCompositeName(final Name name) throws InvalidNameE
             return (CompositeName) name;
         } else {
             final CompositeName compositeName = new CompositeName();
-            compositeName.add(name.toString());
+            compositeName.addAll(new CompositeName(name.toString()));
```

Comment:
Thanks for this explanation. Introducing a new method like toDecomposedCompositeName makes sense. Let me give some context on the two main problems I'm currently seeing:

1) The recent [switch](https://github.com/wildfly/wildfly-naming-client/pull/11) to using NamingUtils.toCompositeName() for version 1 of the remote naming protocol results in a NameNotFoundException when looking up a name like "foo/bar". Because the CompositeName consists of a single segment, the server will end up calling ServiceBasedNamingStore.lookup() using a lookup name of "jboss.naming.context.java.jboss.exported.foo/bar" instead of using "jboss.naming.context.java.jboss.exported.foo.bar". Introducing and making use of new method like toDecomposedCompositeName will fix this issue.

2) The issue above also occurs when using version 2 of the remote naming protocol. In this case, the client [uses](https://github.com/wildfly/wildfly-naming-client/blob/master/src/main/java/org/wildfly/naming/client/remote/RemoteClientTransport.java#L210) ```NamingUtils.toCompositeName(name).toString()``` to form a String name that should be looked up. For a SimpleName with segments "foo" and "bar", the client will use a lookup name of ""foo/bar"" (notice the extra quotes here). Again, the server will end up calling ServiceBasedNamingStore.lookup() using a lookup name of "jboss.naming.context.java.jboss.exported.foo/bar" instead of using "jboss.naming.context.java.jboss.exported.foo.bar". What's the best way to fix this? What's the reason for the client converting the Name to a CompositeName before calling toString()?